manager/integration/tests/test_basic.py

```diff
@@ -3642,6 +3642,165 @@ def test_volume_toomanysnapshots_condition(client, core_api, volume_name):  # NO
                                                "status", "False")
 
 
+@pytest.mark.coretest  # NOQA
+def test_volume_metafile_deleted_empty(client, core_api, volume_name, csi_pv, pvc, pod):  # NOQA
+    """
+    Test volume metafile deleted or empty
+
+    Context:
+
+    If the volume still works fine when there is a invalid volume meta file
+    in the replica data path.
+
+    Setup:
+
+    1. Create a pod using Longhorn volume
+    2. Write some data to the volume then get the md5sum
+    3. Create a snapshot
+
+    Steps:
+
+    1. Delete the pod and wait for the volume detached
+    2. Randomly pick up a replica and manually delete or
+       empty the volume meta file in this replica data path
+    3. Recreate the pod and wait for the volume attached
+    4. Check if the volume is Healthy after the volume attached
+    5. Check volume data
+    6. Check if the volume still works fine by r/w data and
+       creating/removing snapshots
+    """
+    test_pv_name = "pv-" + volume_name
+    test_pvc_name = "pvc-" + volume_name
+    test_pod_name = "pod-" + volume_name
+    volume_size = str(1 * Gi)
+
+    csi_pv['metadata']['name'] = test_pv_name
+    csi_pv['spec']['csi']['volumeHandle'] = volume_name
+    csi_pv['spec']['capacity']['storage'] = volume_size
+    pvc['metadata']['name'] = test_pvc_name
+    pvc['spec']['volumeName'] = test_pv_name
+    pvc['spec']['resources']['requests']['storage'] = volume_size
+    pvc['spec']['storageClassName'] = ''
```

Comment:
Also, try to see if any useful function in common.py instead of duplicating same operations. ex: `prepare_pod_with_data_in_mb`

You can refer to other cases.

Reply:
Replaced by `prepare_pod_with_data_in_mb`, and some snapshot checks. Thanks